xlwings/main.py

```diff
@@ -466,7 +466,23 @@ def name(self):
     @name.setter
     def name(self, value):
         xlplatform.set_worksheet_name(self.xl_sheet, value)
+    
+    @property
```

Comment:
The usage of a getter property for doing something other than retrieving a value seems odd/unusual/surprising. I would rather have a normal method with an optional password argument, so that users can call:

```python
sheet.protect()
sheet.unprotect()
sheet.protect('mypassword')
sheet.unprotect('mypassword')
```

OR, if you really want to use a property, you could have a property for being protected without password but you still need a method to unprotect, so the API would not feel very consistent: 

```python
sheet.protect = True
sheet.protect = False
sheet.password_protect('mypassword')
sheet.password_unprotect('mypassword')
```